framework/addons/__init__.py

```diff
@@ -124,7 +124,8 @@ def delete_addon(self, addon_name, auth=None, _force=False):
             if self._name in addon.config.added_mandatory and not _force:
                 raise ValueError('Cannot delete mandatory add-on.')
             if hasattr(addon, 'external_account'):
-                addon.deauthorize(auth=auth)
+                if getattr(addon, 'external_account'):
```

Comment:
`getattr` takes third argument, which is the default value, if the objects doesn't have the attribute. Therefore, no need to have both `hasattr` and `getattr`.

```python
if getattr(addon, 'external_account', None):
```